Fix Studio Assistant download plugins wp-cli commands by mounting Studio mu-plugins

sejas opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Fixes STU-1126

## Proposed Changes

- Mount Studio mu-plugins before executing a wp-cli command

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**Confirm the issue is fixed**

- Run `npm start`
- Go to the Assistant tab
- Ask to install woocommerce
- Execute the wp-cli command
- Confirm that it works as expected and the plugin is installed
- Go to the mu-plugins folder of your site and confirm there is no new files. Only the `sqlite-database-integration`.

**Try other wp-cli commands like sqlite export**
- Go to the Import/Export tab
- Export your site
- Confirm that the site is correctly exported

| Before | After |
|--------|--------|
|  <img width="1120" height="556" alt="image (10)" src="https://github.com/user-attachments/assets/1054d355-5e2a-4181-98dd-d4295ce84b8d" /> | <img width="849" height="352" alt="Screenshot 2025-12-12 at 10 52 08" src="https://github.com/user-attachments/assets/610f5b9d-1d5f-4339-abec-845dbce18b6c" /> |


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Have you checked for TypeScript, React or other console errors?

sejas committed:
Mount mu plugins when in wp-cli-executor to fix Studio Assistant download URLs

epeicher reviewed (approved):
Thanks for fixing this @sejas! I have tested it, and it works as expected, I can install WooCommerce using the Assistant now. Code changes also LGTM! :shipit: 

|Before | After|WooCommerce installed
|-------|------|------
|<img width="2796" height="1654" alt="Image" src="https://github.com/user-attachments/assets/a2437788-0b86-4975-a726-43a1ef906c27" />|<img width="2796" height="1654" alt="Image" src="https://github.com/user-attachments/assets/39a777cd-f770-4281-bd2d-7cc16832ba88" />|<img width="2356" height="2828" alt="Image" src="https://github.com/user-attachments/assets/42978d0c-8c9f-45e6-9abb-d7ba69c83bb8" />

ivan-ottinger reviewed (approved):
Works great! I can confirm the proposed changes fix the issue. WP CLI plugin installation runs as expected:

<img width="1134" height="1030" alt="Image" src="https://github.com/user-attachments/assets/f6b579f1-4a9a-46bc-9e0e-d5cb58581738" />

and there are no extra files added:

<img width="1134" height="226" alt="Image" src="https://github.com/user-attachments/assets/2391028e-c279-4284-b368-e1f05dd61e5a" />

Site export works as well. 👍🏼 

---

The changes themselves look good to me too. It may be worth having a second pair of eyes look at them since I haven't reviewed many Playground CLI related changes yet and could be missing something there.

wojtekn commented:
@sejas, would you mind sharing more details on what caused the issue and why mounting mu-plugins resolved it?

sejas commented:
> @sejas, would you mind sharing more details on what caused the issue and why mounting mu-plugins resolved it?

@wojtekn, the issue was that some WP-CLI commands didn't work, specifically those that had to download a file from wp.org, such as installing a plugin or theme. We have a mu-plugin, [0-allowed-redirect-hosts.php](https://github.com/Automattic/studio/blob/4a7c181e4cfabf4d00412b60d121fc00040f4fd6/common/lib/mu-plugins.ts#L125-L142), that helps resolving those domains.

When we switched from wp-now to playground-cli, we also changed how Studio mu-plugins are mounted, mounting them only when Studio starts and not when executing wp-cli commands. In this PR, I'm mounting the Studio mu-plugins for wp-cli executions as well. Now, wp-cli executions and Studio start sites will behave more similarly.
